### ISSUE TITLE
[FEATURE REQUEST] Automatically read the LoRA selected in Efficient Loader node and other LoRA loader nodes

### ISSUE THREAD

**Issue by alessandroperilli** [jitcoder/lora-info#3]
This request might be too onerous to satisfy given the enormous amount of LoRA loaders that are out there. 

I'd argue that, at minimum, two main loaders could be supported: the default ComfyUI one, and the `Efficiency Loader` node, which many users adopt to simplify their workflows. 

Or, perhaps, there's a non-resource-intensive way, to read the loaded LoRA model name, no matter what type of node does the operation? 

For example:

The `LoRA Info` node could accept any input coming from a LoRA loading node and, if connected in that way, it would search for the LoRA model information accordingly.

In the following situation, I would link a random output from the `Efficiency Loader` node (let's say `Model`, or `VAE`) to the input of `LoRA Info`, and the latter would do its job.

<img width="505" alt="Screenshot 2023-12-09 at 10 26 54" src="https://github.com/jitcoder/lora-info/assets/9775949/3eab5ff5-d87b-461c-8c17-14c7f42871b6">

I've seen this trick used by rghtree in nodes like Fast Bypasser, so maybe it's possible.

In any way, thank you for considering my request.

**Comment by jitcoder** [comment]
I like this idea a lot. I'll do some research on it. 
the only blocker that is coming to mind is getting non-'outputted' data out of another node, ill mess around though and see if its possible

**Comment by camoody1** [comment]
Check out the Automatic1111 extension called CivitAI Helper. This is EXACTLY what we need and your node is about 75% there!

That extension downloads the CIvitAI information in json form as well as a sample jpg and stores that information with the same model name as the Lora and in the same folder. That way, A1111 doesn't have to keep doing polling to CivitAI every time you want to find Lora info. If you could code YOUR node to look for that same data structure before heading out to Civit, it would be super fast.

I would LOVE if your node could show a sample jpg for each Lora. That's really how I remember what each Lora does. The names are usually pretty awful.

Thank you!

Example of the file structure:
![image](https://github.com/jitcoder/lora-info/assets/73725653/fcadb0d0-d695-46c1-b02a-5e5a35c2e09d)

**Comment by jitcoder** [comment]
> Check out the Automatic1111 extension called CivitAI Helper. This is EXACTLY what we need and your node is about 75% there!
> 
> That extension downloads the CIvitAI information in json form as well as a sample jpg and stores that information with the same model name as the Lora and in the same folder. That way, A1111 doesn't have to keep doing polling to CivitAI every time you want to find Lora info. If you could code YOUR node to look for that same data structure before heading out to Civit, it would be super fast.
> 
> I would LOVE if your node could show a sample jpg for each Lora. That's really how I remember what each Lora does. The names are usually pretty awful.
> 
> Thank you!
> 
> Example of the file structure: ![image](https://private-user-images.githubusercontent.com/73725653/293105389-fcadb0d0-d695-46c1-b02a-5e5a35c2e09d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxOTA2NTksIm5iZiI6MTcwNDE5MDM1OSwicGF0aCI6Ii83MzcyNTY1My8yOTMxMDUzODktZmNhZGIwZDAtZDY5NS00NmMxLWIwMmEtNWU1YTM1YzJlMDlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDEwMTIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1NTI5ZjljZjFhNTU4YTFlMjVmYmQxYjdlMTgzZDdlNGQyZDI2NTUwMGFkNjAxOTQxMDVhNDEwNTE2OTE4MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0xfsCDiDRZiWMeozp1ERjyolzHvqdTcLuJ2rCFA_908)

The node currently already does cache data from civitai, it doesn't actually hit civit every time.
The image preview is a good idea, I'll add that when I get back into comfyui. It might be a while, I've actually deleted comfyui and stable diffusion stuff after getting frustrated with performance (I'm on a M1 Pro Mac), tried the coreml migration stuff but its so buggy and glitchy. taking a break from it for a while, but will definitely be back

**Comment by alessandroperilli** [comment]
@jitcoder I'm on a M2 with 96GB RAM and it takes me 4min to generate an SDXL image of 1344x768px. If I enable any ControlNet (or a FreeU node) even worse. So I know exactly how you feel. 

I'm following closely the development of the Core ML Suite by aszc-dev, but there are too many limitations at the moment. Plus, with the release of the new MLX framework, I'm not sure that effort will continue.

Have a nice break.

**Comment by jitcoder** [comment]
> @jitcoder I'm on a M2 with 96GB RAM and it takes me 4min to generate an SDXL image of 1344x768px. If I enable any ControlNet (or a FreeU node) even worse. So I know exactly how you feel.
> 
> I'm following closely the development of the Core ML Suite by aszc-dev, but there are too many limitations at the moment. Plus, with the release of the new MLX framework, I'm not sure that effort will continue.
> 
> Have a nice break.

yeah hopefully something comes around and makes this more feasible on macos.

**Comment by jitcoder** [comment]
just dropping a small comment here. I haven't forgotten about this request, I'll eventually come to this.
short term I'm going to be working on the preview image feature requested by @camoody1.

I'm also about to drop another node "Load Image From URL", its already done, just testing it out n stuff
